### PR TITLE
Factor up command infra and introduce pex3.

### PR DESCRIPTION
**Pull request by jsirois**
This will clear space for build-time commands and be the new interface for
the Pex CLI in Pex 3.

Work towards #1401.

### CODE REVIEW

File: pex/commands/command.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
```
Comment (jsirois) [inline pex/commands/command.py:1]: Moved from pex/tools/command.py below save for PEXCommand: https://github.com/pantsbuild/pex/pull/1459/files#diff-53a0c89163a0852f29460cfda14daf24bba9612dc102f11888e4b5b1f2d47fb5L149

**Comment by jsirois** [comment]
Not much for now:

```console
$ python -mpex.cli -h
usage: __main__.py [-h] [-V]

Tools for building and working with [P]ython [EX]ecutables..

optional arguments:
  -h, --help     show this help message and exit
  -V, --version  show program's version number and exit
$ python -mpex.cli -V
2.1.50
```

**Comment by jsirois** [comment]
A little thrash there - but g2g.

**Comment by jsirois** [comment]
Let me know if I should loop someone else in for review.

**Review by Eric-Arellano — APPROVED** [review]
Ah sorry, started looking last night and forgot to finish. Exciting!

**Pull request merged**